pootle/apps/pootle_store/forms.py

```diff
@@ -373,3 +407,129 @@ def save(self, **kwargs):
+
+    sfields = forms.MultipleChoiceField(
+        required=False,
+        widget=CommaSeparatedCheckboxSelectMultiple,
+        choices=(
+            ('source', _('Source Text')),
+            ('target', _('Target Text')),
+            ('notes', _('Comments')),
+            ('locations', _('Locations'))),
+        initial=['source', 'target'])
+
+    def __init__(self, *la, **kwa):
+        self.request_user = kwa.pop("user")
+        super(UnitSearchForm, self).__init__(*la, **kwa)
+        self.fields["modified-since"] = ISODateTimeField(required=False)
+
+    def clean_user(self):
+        if self.cleaned_data["user"] is None:
+            if self.request_user.is_anonymous():
+                return get_user_model().get(self.request_user)
+            return self.request_user
+        return self.cleaned_data["user"]
+
+    def clean(self):
+        from .views import ALLOWED_SORTS
+
+        if "checks" in self.errors:
+            del self.errors["checks"]
+            self.cleaned_data["checks"] = None
+        if "user" in self.errors:
+            User = get_user_model()
+            del self.errors["user"]
+            self.cleaned_data["user"] = User.get(self.request_user)
+        if self.errors:
+            return
+        if self.cleaned_data['count'] is None and not self.errors:
+            self.cleaned_data["count"] = (
+                self.cleaned_data["user"].get_unit_rows())
+
+        self.cleaned_data["vfolder"] = None
+        pootle_path = self.cleaned_data.get("path")
+        if 'virtualfolder' in settings.INSTALLED_APPS:
+            vfolder, pootle_path = extract_vfolder_from_path(
+                pootle_path,
+                vfti=VirtualFolderTreeItem.objects.select_related(
+                    "directory", "vfolder"))
+            self.cleaned_data["vfolder"] = vfolder
+            self.cleaned_data["pootle_path"] = pootle_path
+        path_keys = [
+            "project_code", "language_code", "dir_path", "filename"]
+        try:
+            path_kwargs = {
+                k: v
+                for k, v in resolve(pootle_path).kwargs.items()
+                if k in path_keys}
+        except Resolver404:
+            raise forms.ValidationError('Unrecognised path')
+
+        self.cleaned_data.update(path_kwargs)
+
+        sort_on = "units"
+        if "filter" in self.cleaned_data:
+            unit_filter = self.cleaned_data["filter"]
+            if unit_filter in ('suggestions', 'user-suggestions'):
+                sort_on = 'suggestions'
+            elif unit_filter in ('user-submissions', ):
+                sort_on = 'submissions'
+        sort_by_param = self.cleaned_data["sort"]
+
+        if sort_by_param in ALLOWED_SORTS[sort_on]:
+            self.cleaned_data["sort_by"] = ALLOWED_SORTS[sort_on][sort_by_param]
+        else:
+            self.cleaned_data["sort_by"] = None
+
+        self.cleaned_data["sort_on"] = sort_on
+
+    def clean_month(self):
```

Comment:
This is a stupid nit, but I don't like having a three lines method on the bottom because it is easy to miss. Also it kinda belongs with the other `clean_FIELD` methods above.